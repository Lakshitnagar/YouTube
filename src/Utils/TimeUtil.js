export const getVideoDisplayDate = (time) => {
    if(!time) return '';

    const paredDate = new Date(time);

    return paredDate && !isNaN(paredDate.getTime()) ? `${paredDate.getDate()} ${getMonthInWords(paredDate.getMonth())} ${paredDate.getFullYear()}`: '';
};

const getMonthInWords = (month) => {
    let monthInWords = "-";
    // eslint-disable-next-line
    switch (month) {
        case 0:
            monthInWords = "Jan";
            break;
        case 1:
            monthInWords = "Feb";
            break;
        case 2:
            monthInWords = "Mar";
            break;
        case 3:
            monthInWords = "Apr";
            break;
        case 4:
            monthInWords = "May";
            break;
        case 5:
            monthInWords = "June";
            break;
        case 6:
            monthInWords = "July";
            break;
        case 7:
            monthInWords = "Aug";
            break;
        case 8:
            monthInWords = "Sept";
            break;
        case 9:
            monthInWords = "Oct";
            break;
        case 10:
            monthInWords = "Nov";
            break;
        case 11:
            monthInWords = "Dec";
            break;
    }

    return monthInWords;
};
