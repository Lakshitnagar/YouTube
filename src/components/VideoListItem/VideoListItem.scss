.videoListItem{
  height: fit-content;
  margin: 0 8px 40px 8px;
  width: 240px;
  flex-grow: 1;

  .videoListItem__thumbnail{
    width: 100%;
  }

  div{
    padding: 2px 5px;
    max-height: 3rem;
  }

  .videoListItem__title{
    font-size: 1.2rem;
    line-height: 1.5rem;
    overflow: hidden;
    text-overflow: ellipsis;
    margin-top: 10px;
  }

  .videoListItem__channel, .videoListItem__time{
    color: #606060;
  }

  .videoListItem__time{
    font-size: smaller;
  }
}
