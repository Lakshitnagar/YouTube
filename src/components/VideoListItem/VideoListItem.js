import React from 'react';
import './VideoListItem.scss';
import PropTypes from "prop-types";
import {getVideoDisplayDate} from "../../Utils/TimeUtil";

function VideoListItem(props) {
    return (
        <div className="videoListItem">
            <img className="videoListItem__thumbnail" src={props.videoDetail.snippet.thumbnails.medium.url}
                 alt='thumbnail'/>

            <div className="videoListItem__title">{props.videoDetail.snippet.title}</div>
            <div className="videoListItem__channel">{props.videoDetail.snippet.channelTitle}</div>
            <div className="videoListItem__time">{getVideoDisplayDate(props.videoDetail.snippet.publishedAt)}</div>
        </div>
    );
}

export default VideoListItem;

VideoListItem.propTypes = {
    videoDetail: PropTypes.object,
};
