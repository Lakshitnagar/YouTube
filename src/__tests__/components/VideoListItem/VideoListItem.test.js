import React from 'react';
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import VideoListItem from "../../../components/VideoListItem/VideoListItem";
import * as TimeUtil from "../../../Utils/TimeUtil";

describe('VideoListItem', function () {
    beforeEach(()=>{
        spyOn(TimeUtil, 'getVideoDisplayDate').and.returnValue('DISPLAY_DATE');
    });
    it('should render and match snapshot', function () {
        const videoDetail = {
            snippet: {
                thumbnails: {
                    medium: {url: 'THUMBNAIL_URL'}
                },
                title: 'VIDEO_TITLE',
                channelTitle: 'CHANNEL_TITLE',
                publishedAt: 'PUBLISHED_AT'
            }
        };

        const component = <VideoListItem videoDetail={videoDetail}/>;

        const wrapper = shallow(component);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(TimeUtil.getVideoDisplayDate).toHaveBeenCalledWith('PUBLISHED_AT');
    });
});
