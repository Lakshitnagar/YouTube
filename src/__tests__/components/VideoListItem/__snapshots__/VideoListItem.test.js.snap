// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoListItem should render and match snapshot 1`] = `
<div
  className="videoListItem"
>
  <img
    alt="thumbnail"
    className="videoListItem__thumbnail"
    src="THUMBNAIL_URL"
  />
  <div
    className="videoListItem__title"
  >
    VIDEO_TITLE
  </div>
  <div
    className="videoListItem__channel"
  >
    CHANNEL_TITLE
  </div>
  <div
    className="videoListItem__time"
  >
    DISPLAY_DATE
  </div>
</div>
`;
